.github/workflows/publish.yml

```
@@ -0,0 +1,78 @@
+  test:
+    call-shared:
+      uses: ./.github/workflows/shared-workflow.yml
+
+  detect-environments:
+    runs-on: ubuntu-latest
+    needs: [test, lint]
+    outputs:
+      environments: ${{ steps.environments.outputs.result }}
+    if: github.ref_name == 'main'
+    steps:
+      - uses: actions/github-script@v6
```

actionlint (1.7.7)

the runner of "actions/github-script@v6" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)